frontend/scripts/react-components/data-portal/data-portal-page.component.jsx

```diff
@@ -121,17 +134,16 @@ class DataContent extends Component {
         const selectedContext = this.props.contexts.find(
           elem => elem.isDisabled !== true && elem.id === value
         );
-        this.setState({ selectedCommodity: value, selectedContextId: selectedContext.id });
+        this.setState({ selectedCommodity: value });
+        this.props.selectContextById(selectedContext.id);
```

Comment:
Can `selectedContext` be null here? Also what is the difference between `props.selectContextById` and `props.onContextSelected` looks like duplication for me.